src/cbc_sdk/workload/vulnerability_assessment.py

```diff
@@ -0,0 +1,902 @@
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+        """
+        self._doc_class = doc_class
+        self._cb = cb
+        self._count_valid = False
+        super(BaseQuery, self).__init__()
+        self._query_builder = QueryBuilder()
+        self._criteria = {}
+        self._sortcriteria = {}
+        self._total_results = 0
+
+    def _build_url(self, tail_end):
+        """
+        Creates the URL to be used for an API call.
+
+        Args:
+            tail_end (str): String to be appended to the end of the generated URL.
+
+        Returns:
+            str: The complete URL.
+        """
+        urlobject = '/vulnerability/assessment/api/v1/orgs/{}'.format(self._cb.credentials.org_key)
+        if self._doc_class._vcenter_specific:
+            urlobject += "/vcenters/{}".format(self._doc_class._vcenter_specific)
+        urlobject += '/devices/{}/vulnerabilities'.format(self._doc_class._model_unique_id)
+        url = urlobject + tail_end
+        return url
+
+    def _build_request(self, from_row, max_rows, add_sort=True):
+        """
+        Creates the request body for an API call.
+
+        Args:
+            from_row (int): The row to start the query at.
+            max_rows (int): The maximum number of rows to be returned.
+            add_sort (bool): If True(default), the sort criteria will be added as part of the request.
+
+        Returns:
+            dict: The complete request body.
+        """
+        request = {
+            "criteria": self._criteria,
+            "query": self._query_builder._collapse(),
+            "rows": 100}
+        # Fetch 100 rows per page (instead of 10 by default) for better performance
+        if from_row > 0:
+            request["start"] = from_row
+        if max_rows >= 0:
+            request["rows"] = max_rows
+        if add_sort and self._sortcriteria != {}:
+            request["sort"] = [self._sortcriteria]
+        return request
+
+    def _perform_query(self, from_row=0, max_rows=-1):
+        """
+        Performs the query and returns the results of the query in an iterable fashion.
+
+        Args:
+            from_row (int): The row to start the query at (default 0).
+            max_rows (int): The maximum number of rows to be returned (default -1, meaning "all").
+
+        Returns:
+            Iterable: The iterated query.
+        """
+        url = self._build_url("/_search")
+        current = from_row
+        numrows = 0
+        still_querying = True
+        while still_querying:
+            request = self._build_request(current, max_rows)
+            resp = self._cb.post_object(url, body=request)
+            result = resp.json()
+
+            self._total_results = result["num_found"]
+            self._count_valid = True
+
+            results = result.get("results", [])
+            for item in results:
+                yield item
```

Comment:
We should be building the DeviceVulnerability model similar to how we do it here https://github.com/carbonblack/carbon-black-cloud-sdk-python/blob/develop/src/cbc_sdk/platform/devices.py#L507